src/main/java/org/kohsuke/github/GHBranchProtectionBuilder.java

```diff
@@ -532,7 +564,7 @@ private Restrictions getRestrictions() {
 
     private StatusChecks getStatusChecks() {
         if (statusChecks == null) {
-            statusChecks = new StatusChecks();
+            statusChecks = new StatusChecksWithAppId();
         }
         return statusChecks;
     }
```

Comment:
Suggest something like this:
```
    <T extends StatusCheck> T getStatusChecks(Class<T> clazz) {
        if (statusChecks == null) {
            try {
                statusChecks = clazz.getDeclaredConstructor().newInstance();
            } catch (Exception ignored) {
            }
        }
        if (!clazz.isInstance(statusChecks) {
            throw new GHException("Cannot use checks and context status checks");
        } 
        return (T) statusChecks;
    }
```

Reply:
Apparently the deprecated API can be implemented in the new API. From [the docs](https://docs.github.com/en/enterprise-server@3.0/rest/reference/branches#update-branch-protection--parameters):
![image](https://github.com/hub4j/github-api/assets/61763026/ce489ff9-da66-41fc-9224-369a1b205039)

I underlined the same logic between the old and new API.

I.e. you need to omit the `app_id` field to do it the old way. I've added the `@JsonInclude(JsonInclude.Include.NON_NULL)` to instruct jackson to omit it in that case, and a comment explaining the special "app_id" values.

So I committed a change which allows the use of both the old and new user methods at the same time, preserving both their effects.

How does that look to you?